src/validation.cpp

```diff
@@ -1331,6 +1322,37 @@ MempoolAcceptResult MemPoolAccept::AcceptSingleTransaction(const CTransactionRef
         return MempoolAcceptResult::Failure(ws.m_state);
     }
 
+    if (m_subpackage.m_rbf && !ReplacementChecks(ws)) {
+        if (ws.m_state.GetResult() == TxValidationResult::TX_RECONSIDERABLE) {
+            // Failed for incentives-based fee reasons. Provide the effective feerate and which tx was included.
+            return MempoolAcceptResult::FeeFailure(ws.m_state, CFeeRate(ws.m_modified_fees, ws.m_vsize), single_wtxid);
+        }
+        return MempoolAcceptResult::Failure(ws.m_state);
+    }
+
+    // Check if the transaction would exceed the cluster size limit.
+    if (!m_subpackage.m_changeset->CheckMemPoolPolicyLimits()) {
+        ws.m_state.Invalid(TxValidationResult::TX_MEMPOOL_POLICY, "too-large-cluster", "");
+        return MempoolAcceptResult::Failure(ws.m_state);
+    }
+
```

Comment:
00c02b42c6199a6c46573304761056082490c7d4

While we're here: Suggestion to make it more clear as to the roles of the various TRUC checks and that everything works once dependencies are attached.

```Suggestion
    // Either sibling eviction was needed, or package based check should always pass for single tx given SingleTRUCChecks already passed
    Assume(ws.m_sibling_eviction || !PackageTRUCChecks(m_pool, ws.m_ptx, ws.m_vsize, {ws.m_ptx}, ws.m_parents));
```